### DISCUSSION TITLE
Slow scan ... is cache being used?

### DISCUSSION THREAD

**Discussion by BEisem** [Taxel/PlexTraktSync#469] [Q&A] [answered]
Hey all, I am trying out the new version (currently on 0.13.21). I have started from scratch with a fresh install. I'm finding that the scans are taking a very long time, even after the cache has been built. Keep in mind, I have a very large library, and all my data is stored on a cloud drive. But I would expect that once the info is cached into `trakt_cache.sqlite`, it should go much quicker. But that doesn't seem to be happening.

I'm getting about 1 it/s (I assume that means 1 item per second) even after the initial scan. It is currently estimating about 12 hours to complete the scan. That just doesn't seem right, and is much much longer than previous versions, which only took about an hour.

I noticed a new entry in config.json, which is:

```
    "cache": {
        "path": "$PTS_CACHE_DIR/trakt_cache"
    },
```

I didn't change that, but I'm wondering if I need to adjust that in the config? In any event, I'm sure I'm doing something wrong and was hoping someone could point me in the right direction. Thanks!

**Comment by glensc** [comment, marked as answer]
It's a balance of accuracy vs performance. And accuracy is more important to people:
- https://github.com/Taxel/PlexTraktSync/pull/466

**Comment by BEisem** [comment]
Ok, I agree, accuracy is important. But does `episodes()` apply for TV shows only, or also for movies? It seems my movies scan is very slow as well.

**Reply by glensc** [reply]
#466 affects episodes, however, there's an earlier fix for movies https://github.com/Taxel/PlexTraktSync/pull/410 which has the same effect.

you can go to plex forums and vote that they improve their APIs:
- https://github.com/pkkid/python-plexapi/issues/808#issuecomment-897316843

media items indexed with the legacy agent doesn't have such extra request need for each media item.

**Comment by BEisem** [comment]
I just saw the release notes for the new Plex server version, and it has `(API) Added 'includeGuids' query param option (#12976)` as a fix.... if I understand correctly, it looks like they added this!

**Reply by simonc56** [reply]
Yes, very good news !
[confirmed](https://forums.plex.tv/t/implemented-add-includeguids-url-parameter-to-return-external-guids/735620/106) in Plex Media Server [1.24.3.5033](https://forums.plex.tv/t/plex-media-server/30447/462) (beta version)

**Reply by glensc** [reply]
Plex API already has a pr:
- https://github.com/pkkid/python-plexapi/pull/831

So, just need to upgrade dependencies with it and use new pms.

however, this would solve only guids slowness, for episodes there's still need to retrieve allLeaves for each show.
